flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql/SqlModelCall.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.calcite.sql;
+
+import org.apache.flink.table.planner.catalog.CatalogSchemaModel;
+import org.apache.flink.table.planner.plan.FlinkCalciteCatalogReader;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql.validate.SqlValidatorCatalogReader;
+import org.apache.calcite.sql.validate.SqlValidatorScope;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import static org.apache.calcite.util.Static.RESOURCE;
+
+/** SqlModelCall to fetch and reference model based on identifier. */
+public class SqlModelCall extends SqlBasicCall {
+
+    private @Nullable CatalogSchemaModel model = null;
+
+    public SqlModelCall(SqlExplicitModelCall modelCall) {
+        super(
+                SqlModelOperator.create(
```

Comment:
We can introduce a static field for this operator, e.g.
```
    private static final SqlOperator OPERATOR = new SqlModelCallOperator();

```

Reply:
Good point!